File: python/examples/generate_token.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

"""Generate a token without connecting to a cache."""

from developer_toolkit_elasticache import generate_iam_auth_token

token = generate_iam_auth_token(
    serverless_cache_name="my-cache",
    user_id="my-user",
    region="us-east-1",
)

print("Token generated successfully!\n")
print(f"Token length: {len(token)} chars")
print(f"Token preview: {token[:80]}...")

File: node/examples/generate-token.mjs
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/* global console */

import { generateIamAuthToken } from "@aws/developer-toolkit-elasticache";

const token = await generateIamAuthToken({
  serverlessCacheName: "my-cache",
  userId: "my-iam-user",
  region: "us-east-1",
});

// Do not print the token. It is a short-lived bearer credential.
console.log(`Generated an IAM auth token (${token.length} characters).`);
